test/ecto/changeset_test.exs

```diff
@@ -1127,6 +1127,10 @@ defmodule Ecto.ChangesetTest do
                 |> validate_number(:decimal, equal_to: Decimal.new(-1))
     assert changeset.valid?
 
+    changeset = changeset(%{"decimal" => Decimal.new(0)})
+                |> validate_number(:decimal, not_equal_to: Decimal.new(-1))
+    assert changeset.valid?
```

Comment:
This test case is only checking things for decimals but we need a regular test as well, that checks that `validations(changeset)` returns the proper metadata and what happens in case of errors. Thanks!

Reply:
Added simple test to check whether metadata are correct. However in the process I have found possible bug, but I am not sure about that. To be exact, data set as `default` aren't checked by the validations, so this will pass:

```elixir
defmodule Foo do
  use Ecto.Schema

  schema "foos" do
    field :count, :integer, default: 0
  end

  def changeset(params) do
    %__MODULE__{}
    |> validate_number(:count, not_equal_to: 0)
  end
end

assert Foo.changeset(%{"count" => 0}).valid?
```